meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-mender/tegra-state-scripts/files/tegra194/switch-rootfs

```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+echo "$(mender show-artifact): $(basename "$0") was called!" >&2
+
+get_bootpart() {
+	local current_slot=`nvbootctrl get-current-slot 2>/dev/null`
+	if [ -z "$current_slot" ]; then
+    		echo "ERR: could not identify current boot slot" >&2
+    		echo "UNKNOWN"
+    		return
+    	fi
+	echo "$current_slot"
+}
+
+reset_unbootable_status() {
+	local AB="AB"
+	local slot="Slot${AB:$1:1}"
+	local varname="/opt/nvidia/esp/EFI/NVDA/Variables/RootfsStatus${slot}-781e084c-a330-417c-b678-38e696380cb9"
+
+	local f=$(mktemp)
+	printf "\x07\x00\x00\x00\x00\x00\x00\x00" > $f
+	dd if=$f of=$varname
+	rm ${f}
+}
+
+
+current_slot=`get_bootpart`
+echo "current_slot=$current_slot" >&2
+
+next_slot=
+if [ $current_slot = "0" ]; then
+	next_slot="1"
+else
+	next_slot="0"
+fi
+
+echo "next_slot=$next_slot" >&2
+nvbootctrl set-active-boot-slot $next_slot
```

Comment:
See https://github.com/OE4T/meta-tegra/pull/1285#pullrequestreview-1547992166 - I think the only thing you need here for capsule updte support is to an RDEPENDS on `tegra-uefi-capsules` and a call to `oe4t-set-uefi-OSIndications` here.

Reply:
That's great if that is all, will attempt!

Reply:
Also regarding the RDEPENDS in `tegra-state-scripts` it looks like the dependencies for mender are currently added [here](https://github.com/OE4T/meta-mender-community/blob/22f8768ffc41839695a9c8b536caf20ef9340a96/meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-mender/mender-client/mender-client_%25.bbappend#L1-L4)

Reply:
I added the copy as described in the linked PR as well, but now I'm getting an error during boot:
`Mar 05 09:09:06 benchtop-neo-1 nvbootctrl[786]: Info: switch boot chain failed, variable name: BootChainFwStatus, value: 3.` (tested on Orin). 

I boot up in the same partition as before, so the upgrade failed. If I remove the `oe4t-set-uefi-OSIndications` it works again.

I have removed the capsule updates from the switch-rootfs script for now, will investigate more when I'm back. 
Again, let me know if you would prefer to close this MR, merge it or just leave it open. I will keep on investigating the two open items once I'm back anyway.

@dwalkes

Reply:
@mwest90 I'm fine with leaving it open.  I would like to try it myself so either a `tegra-demo-distro` fork or some instructions about the setup you've got would be great.  I think upstream mender is going to kas so we should probably update kas config and instructions before we attempt to upstream.  That doesn't necessarily need to be in the scope of this PR though.

Reply:
@mwest90 @dwalkes I have just rewritten this script to switch rootfs using capsule from new rootfs. It works for my Jetson Orin Nano. Would you like me to add it?

Regarding the RDEPENDS in tegra-state-scripts added [here](https://github.com/OE4T/meta-mender-community/blob/22f8768ffc41839695a9c8b536caf20ef9340a96/meta-mender-tegra/meta-mender-tegra-jetpack5/recipes-mender/mender-client/mender-client_%25.bbappend#L1-L4). Which overrides from `tegra`, `tegra210` and `jetson-agx-orin-devkit` should work for meta-tegra-jetpack5

Reply:
@irodriguez-veridas would be great if you could add it in my opinion!

Reply:
Should I PR to your repo @mwest90 so that you can add it to this PR?

Reply:
@irodriguez-veridas sounds good

Reply:
@mwest90 PR added: https://github.com/mwest90/meta-mender-community/pull/1

Reply:
@irodriguez-veridas I just tested with Xavier now, and this works. 
I have one concern however.
Sine the "reset unbootable status" function was removed, will we be able to get out of the case where the bootloader marks a partition as "bad"?  Is setting the uefi indications enough to boot back into the "bad" partition after it has been re-flashed?

Reply:
You are totally right, sorry, I misunderstood previous comment and expected `oe4t-set-uefi-OSIndications` to do all the work. Yesterday I finally tested updating to unbootable slot and it misses clearing the the unbootable flag. I have reverted this part and added PR with it: https://github.com/mwest90/meta-mender-community/pull/6

Reply:
@irodriguez-veridas thank you so much!  Merged it back in.